src/main/java/org/example/dao/UserDAOImpl.java

```diff
@@ -0,0 +1,55 @@
+package org.example.dao;
+
+import lombok.RequiredArgsConstructor;
+import org.example.entities.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@Repository
+public class UserDAOImpl implements UserDAO {
+
+    private final SessionFactory sessionFactory;
+
+    @Override
+    @Transactional
+    public void save(User user) {
+        Session session = sessionFactory.getCurrentSession();
+        session.persist(user);
+    }
+
+    @Override
+    @Transactional
+    public User findById(int id) {
+        Session session = sessionFactory.getCurrentSession();
+        return session.get(User.class, id);
+    }
```

Comment:
Don't need Transaction annotation when you make GET request. Only when you change the data in database you need to use Transactional